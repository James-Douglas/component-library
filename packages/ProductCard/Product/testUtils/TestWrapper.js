import React from 'react';
import ProductProvider from '../../Context/ProductProvider';

const ProductTestWrapper = ({
  // eslint-disable-next-line react/prop-types
  children, product, onDetails = () => {}, onSelect = () => {},
}) => (
  <ProductProvider product={product} onDetails={onDetails} onSelect={onSelect}>
    { children }
  </ProductProvider>
);

export default ProductTestWrapper;
