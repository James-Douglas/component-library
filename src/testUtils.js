// test-utils.js
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './providers/ManorProvider';

const customRender = (ui, options) => render(ui, { wrapper: ManorProvider, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
