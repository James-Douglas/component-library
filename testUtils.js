import React from 'react';
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './packages/Provider/ManorProvider';

const ManorTestWrapper = ({ children }) => (
  <ManorProvider>
    {children}
  </ManorProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: ManorTestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
