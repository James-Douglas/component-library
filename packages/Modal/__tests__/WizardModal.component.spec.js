import React from 'react';
import 'jest-styled-components';
import { render } from '../../../testUtils';
import WizardModal from '../WizardModal.component';

/* Modal
–––––––––––––––––––––––––––––––––––––––––––––––––– */

describe('WizardModal', () => {
  it('renders with minimal props', () => {
    const { baseElement } = render(<WizardModal id="test-wizard-modal" visible trackingLabel="test-wizard-modal-trackingLabel" />);
    const wizardModal = baseElement.querySelector('#test-wizard-modal');
    expect(wizardModal).toBeInTheDocument();
    expect(wizardModal).toMatchSnapshot();
  });

  it('does not render if visible={false}', () => {
    const { container } = render(<WizardModal id="test-wizard-modal" visible={false} trackingLabel="test-wizard-modal-trackingLabel" />);
    const modal = container.querySelector('#test-wizard-modal');
    expect(modal).not.toBeInTheDocument();
  });

  it('renders children and renders other components', () => {
    const { baseElement } = render(
      <WizardModal id="test-modal" visible trackingLabel="test-wizard-modal-trackingLabel">
        <div id="test-content">
          <p>some test text</p>
        </div>
        <div id="test-content2">
          <p>some other test text</p>
        </div>
      </WizardModal>,
    );

    const modal = baseElement.querySelector('#test-modal');
    const testDiv = modal.querySelector('#test-content');
    const testDiv2 = modal.querySelector('#test-content2');

    expect(testDiv2).toBeInTheDocument();
    expect(testDiv).toBeInTheDocument();
    expect(modal).toBeInTheDocument();
  });
});
