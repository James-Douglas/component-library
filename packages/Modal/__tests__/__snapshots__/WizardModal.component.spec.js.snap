// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardModal renders with minimal props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #f6f6f6;
  box-shadow: 0 1.5rem 3rem 0 rgba(0,0,0,0.11),0 0.5rem 1.5rem 0 rgba(0,0,0,0.08);
  z-index: inherit;
  max-width: 76.8rem;
  min-width: 32.5rem;
  height: 100%;
  max-height: 69.6rem;
  pointer-events: auto;
  position: relative;
  -webkit-animation-name: jkFcyt;
  animation-name: jkFcyt;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 2.4rem 2.4rem 2rem 2.4rem;
}

.c2 {
  cursor: pointer;
  z-index: inherit;
  color: #757575;
  font-size: 2.4rem;
}

<div
  class="c0 
    
  "
  id="test-wizard-modal"
>
  <div
    class="c1"
  >
    <div
      aria-label="Close Modal"
      aria-pressed="false"
      class="c2 icon-close"
      role="button"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-times "
        data-icon="times"
        data-prefix="far"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M207.6 256l107.72-107.72c6.23-6.23 6.23-16.34 0-22.58l-25.03-25.03c-6.23-6.23-16.34-6.23-22.58 0L160 208.4 52.28 100.68c-6.23-6.23-16.34-6.23-22.58 0L4.68 125.7c-6.23 6.23-6.23 16.34 0 22.58L112.4 256 4.68 363.72c-6.23 6.23-6.23 16.34 0 22.58l25.03 25.03c6.23 6.23 16.34 6.23 22.58 0L160 303.6l107.72 107.72c6.23 6.23 16.34 6.23 22.58 0l25.03-25.03c6.23-6.23 6.23-16.34 0-22.58L207.6 256z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;
